torch/nn/modules/rnn.py

```diff
@@ -453,17 +464,19 @@ class RNN(RNNBase):
         >>> output, hn = rnn(input, h0)
     """
 
-    def __init__(self, *args, **kwargs):
-        if 'proj_size' in kwargs:
-            raise ValueError("proj_size argument is only supported for LSTM, not RNN or GRU")
-        self.nonlinearity = kwargs.pop('nonlinearity', 'tanh')
+    def __init__(self, input_size: int, hidden_size: int, num_layers: int = 1,
```

Comment:
Hey, many apologies for not catching this earlier but unfortunately, this change from `*args, **kwargs` --> the actual parameters might be BC breaking so we cannot make this change. :( 

To make the documentation render properly and fix https://github.com/pytorch/pytorch/issues/103845, we could instead add the parameter list to the first line of the docstring. (e.g. [what was done here](https://www.internalfb.com/code/fbsource/[d14d9248df3cdb4820ac0c1eead86935f6970179]/fbcode/caffe2/torch/nn/functional.py?lines=34)) should make the docs render the argument list correctly.

Would you be willing to make this change?

Further, to get the typing correct you can do define an `@overload` above 

e.g.

```
@overload
def forward(self, input_size: int, hidden_size: int, num_layers: int = 1,. etc.):
      ...

# leave the current forward untouched
def forward(*args, **kwargs):
```

Reply:
Not a problem! Thank you so much for clarifying! I'll make this change first thing tomorrow morning and mark the conversation as resolved when I'm done. I really appreciate your help :)

Reply:
Hey, sorry for the delay! 

I think it should be ok to keep the parameter list in one line, appreciate the effort adding the decorator though!

 For example, we definitely break the 80 character limit (e.g. [here](https://github.com/pytorch/pytorch/blob/main/torch/nn/functional.py#L3538)). You can push a commit to see whether CI will error (perhaps we will need to add a noqa or something to the like).

Reply:
Wonderful! Thank you so much for your help! I made the changes with my most recent commit and I'll keep an eye out for CI to see if it needs a noqa. 

This was my first PR request, so your help and kindness were very much appreciated :)